File: code/library/src/main/java/org/fbme/lib/iec61499/stringify/SubappTypePrinter.java
package org.fbme.lib.iec61499.stringify;

import org.fbme.lib.iec61499.declarations.SubapplicationTypeDeclaration;
import org.jdom.Element;
import org.jetbrains.annotations.NotNull;

public class SubappTypePrinter extends DeclarationPrinterBase<SubapplicationTypeDeclaration> {

    public SubappTypePrinter(SubapplicationTypeDeclaration declaration) {
        super(declaration, "SubAppType");
    }

    @Override
    protected void printDeclarationBody(@NotNull Element element) {
        element.addContent(new FBInterfacePrinter(myElement, true).print());
        element.addContent(new SubappNetworkPrinter(myElement.getNetwork()).print());
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/stringify/SubappTypePrinter.kt
package org.fbme.lib.iec61499.stringify

import org.fbme.lib.iec61499.declarations.SubapplicationTypeDeclaration
import org.jdom.Element

class SubappTypePrinter(declaration: SubapplicationTypeDeclaration) :
    DeclarationPrinterBase<SubapplicationTypeDeclaration>(declaration, "SubAppType") {
    override fun printDeclarationBody(element: Element) {
        element.addContent(FBInterfacePrinter(this.element, true).print())
        element.addContent(SubappNetworkPrinter(this.element.network).print())
    }
}
